src/main/java/org/springframework/data/couchbase/core/query/OptionsBuilder.java

```diff
@@ -110,8 +116,17 @@ public static TransactionQueryOptions buildTransactionQueryOptions(QueryOptions
 			throw new IllegalArgumentException("QueryOptions.flexIndex is not supported in a transaction");
 		}
 
+		Object value = optsJson.get("args");
+		if(value instanceof JsonObject){
+			txOptions.parameters((JsonObject)value);
+		}else if(value instanceof JsonArray) {
+			txOptions.parameters((JsonArray) value);
+		}
+
 		for (Map.Entry<String, Object> entry : optsJson.toMap().entrySet()) {
-			txOptions.raw(entry.getKey(), entry.getValue());
+			if(!entry.getKey().equals("args")) {
+				txOptions.raw(entry.getKey(), entry.getValue());
```

Comment:
Continue to not feel good about this approach to clone a QueryOptions into a TransactionQueryOptions...  It won't handle things like the JsonSerializer, for example.